src/main/java/com/sk89q/craftbook/mechanics/HiddenSwitch.java

```diff
@@ -145,13 +154,15 @@ public void onRightClick(PlayerInteractEvent event) {
             event.setCancelled(true);
     }
 
-    private static void toggleSwitches(Block sign, BlockFace direction) {
+    private static void toggleSwitches(Block signBlock) {
```

Comment:
Is there a reason you’re grabbing it from the sign rather than passing it through? Passing it through means one less allocation of a state object

Reply:
The `direction` here is the orientation of the clicked block face, not the direction in which the sign is facing. So in my picture if you were to break the podzol block and then click on that side of the netherrack, the lever on the polished granite would not flip. That's because the face you click is oriented east-west, and so the code would search east-west around the sign and not find anything. So the direction needs to be calculated with the orientation of the sign itself, not the face of the attached block that you click.